lyse/__init__.py

```diff
@@ -244,48 +244,15 @@ def __init__(self,h5_path,no_write=False):
         if not self.no_write:
             self._create_group_if_not_exists(h5_path, '/', 'results')
                      
-        try:
-            if not self.no_write:
-                # The group where this run's results will be stored in the h5
-                # file will be the name of the python script which is
-                # instantiating this Run object. Iterate from innermost caller
-                # to outermost. The name of the script will be one frame in
-                # from analysis_subprocess.py.
-                analysis_subprocess_path = os.path.join(
-                    LYSE_DIR,
-                    'analysis_subprocess.py',
-                )
-                group = None
-                inner_frame = inspect.currentframe()
-                inner_path = self._frame_to_path(inner_frame)
-                inner_file_name = self._path_to_file_name(inner_path)
-                while group is None:
-                    # self._frame_to_path() will raise a KeyError if this loop
-                    # reaches the outermost caller.
-                    outer_frame = inner_frame.f_back
-                    outer_path = self._frame_to_path(outer_frame)
-                    outer_file_name = self._path_to_file_name(outer_path)
-                    if outer_path == analysis_subprocess_path:
-                        group = inner_file_name
-                    inner_frame = outer_frame
-                    inner_path = outer_path
-                    inner_file_name = outer_file_name
-                self.set_group(group)
-        except KeyError:
-            # sys.stderr.write('Warning: to write results, call '
-            # 'Run.set_group(groupname), specifying the name of the group '
-            # 'you would like to save results to. This normally comes from '
-            # 'the filename of your script, but since you\'re in interactive '
-            # 'mode, there is no script name.\n')
-            pass
-
-    def _frame_to_path(self, frame):
-        path = frame.f_globals['__file__']
-        return path
-
-    def _path_to_file_name(self, path):
-        file_name = os.path.basename(path).split('.py')[0]
-        return file_name
+        if not self.no_write:
```

Comment:
These if statements could be combined now.

Reply:
Excellent, -1 additional line, I love deleting code